reviewing.md

```diff
@@ -0,0 +1,126 @@
+- [The Gentle Art Of Patch Review](https://sage.thesharps.us/2014/09/01/the-gentle-art-of-patch-review/)
+
+## Definition
+
+(from [Code Review Developer Guide - Google](https://google.github.io/eng-practices/review/)) 
+
+A code review is a process where someone other than the author(s) of a piece of code examines that code
+
+Code reviews should look at:
+
+- Design: Is the code well-designed and appropriate for your system?
+- Functionality: Does the code behave as the author likely intended? Is the way the code behaves good for its users?
+- Complexity: Could the code be made simpler? Would another developer be able to easily understand and use this code
+  when they come across it in the future?
+- Tests: Does the code have correct and well-designed automated tests?
+- Naming: Did the developer choose clear names for variables, classes, methods, etc.?
+- Comments: Are the comments clear and useful?
+- Style: Does the code follow our style guides?
+- Documentation: Did the developer also update relevant documentation?
+
+## CAPI Design considerations
+
+Within the context of cluster API the following design items should be carefully evaluated when reviewing a PR:
+
+### Reentrancy
+
+In CAPI most of the coding activities happen in controllers, and in order to make robust controllers,
+we should strive for implementing reentrant code.
+
+A reentrant code can be interrupted in the middle of its execution and then safely be called again
+("re-entered"); this concept, applied to Kubernetes controllers, means that a controller should be capable
+of recovering from interruptions, observe the current state of things, and act accordingly. e.g.
+ 
+- We should not rely on flags/conditions from previous reconciliations since we are the controller
+  setting the conditions. Instead, we should detect the status of things through introspection at
+  every reconciliation and act accordingly.
+- It is acceptable to rely on status flags/conditions e that we've previously set as part
+  of the current reconciliation.
+- It is acceptable to rely on status flags/conditions set by other controllers.
+
+NOTE: An important use case for reentrancy is the move operation, where Cluster API objects gets moved
+to a different management cluster and the controller running on the target cluster has to
+rebuild the object status from scratch by observing the current state of the underlying infrastructure.
+
+### API design
+
+The API defines the main contract with the Cluster API users. As most of the APIs in Kubernetes,
+each API version encompasses a set of guarantees to the user in terms of support window, stability,
+and upgradability. 
+
+This makes API design a critical part of Cluster API development and usually:
+
+- Breaking/major API changes should go through the CAEP process and be strictly synchronized with the major
+  release cadence.
+- Non-breaking/minor API changes can go in minor releases.
```

Comment:
Should this add in that non-breaking changes are generally:
- additive in nature
- default to pre-existing behavior

And if it involves crossing multiple layers:
- optional as part of the API contract